### PR TITLE
Bug fix/waiting list unstake edge case

### PR DESCRIPTION
**Pull request by sasurobert**
there was a discrepancy of data between auction and staking smart contract data of waiting and staking bls keys. The auction contract does not know when one node becomes staked instead of waiting.

### CODE REVIEW

File: vm/systemSmartContracts/governance.go [hunk 1]
```diff
@@ -782,30 +781,37 @@ func (g *governanceContract) revokeVotePower(_ *vmcommon.ContractCallInput) vmco
...
-	vmOutput, err := g.executeOnAuctionSC([]byte(txData))
-	if err != nil {
-		return 0, err
-	}
-	if vmOutput.ReturnCode != vmcommon.Ok {
-		return 0, vm.ErrNotEnoughQualifiedNodes
-	}
-	if len(vmOutput.ReturnData) == 0 {
-		return 0, vm.ErrNotEnoughQualifiedNodes
+	marshalledData := g.eei.GetStorageFromAddress(g.auctionSCAddress, address)
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/governance.go:785]: marshaledData *

File: vm/systemSmartContracts/governance.go [hunk 2]
```diff
@@ -782,30 +781,37 @@ func (g *governanceContract) revokeVotePower(_ *vmcommon.ContractCallInput) vmco
...
 
 	auctionData := &AuctionData{}
-	err = g.marshalizer.Unmarshal(auctionData, vmOutput.ReturnData[0])
+	err := g.marshalizer.Unmarshal(auctionData, marshalledData)
 	if err != nil {
 		return 0, err
 	}
 
-	return auctionData.NumStaked, nil
+	numStakedNodes := uint32(0)
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/governance.go:796]: numStakedNodes is not used

File: vm/systemSmartContracts/stakingWithWaiting_test.go [hunk 3]
```diff
@@ -139,3 +139,96 @@ func TestStakingWaitingSC_ExecuteStakeStakeWaitingUnBondFromWaiting(t *testing.T
 	marshalledData = eei.GetStorage(blsKey2)
 	assert.Equal(t, 0, len(marshalledData))
 }
+
+func TestStakingWaitingSC_ExecuteStakeStakeUnStakeStakeUnstake(t *testing.T) {
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/stakingWithWaiting_test.go:143]: :+1: for explaining the reasoning behind the test

File: vm/systemSmartContracts/governance_test.go [hunk 4]
```diff
@@ -476,9 +485,15 @@ func testExecuteVote(t *testing.T, vote []byte) {
...
 			return generalProposalBytes
 		},
+		GetStorageFromAddressCalled: func(address []byte, key []byte) []byte {
+			if bytes.Equal(key, validatorAddr) {
+				return auctionDataBytes
+			}
+			return nodeDataBytes
+		},
 		ExecuteOnDestContextCalled: func(destination, sender []byte, value *big.Int, input []byte) (output *vmcommon.VMOutput, err error) {
 			autionData := &AuctionData{
```
Comment (BeniaminDrasovean) [inline vm/systemSmartContracts/governance_test.go:495]: Use = instead of :=

File: vm/systemSmartContracts/governance_test.go [hunk 5]
```diff
@@ -476,9 +485,15 @@ func testExecuteVote(t *testing.T, vote []byte) {
...
-				NumStaked: 1,
+				NumRegistered: 1,
 			}
 			auctionDataBytes, _ := json.Marshal(autionData)
```
Comment (BeniaminDrasovean) [inline vm/systemSmartContracts/governance_test.go:498]: Use = instead of :=

File: vm/systemSmartContracts/eei.go [hunk 6]
```diff
@@ -67,26 +67,30 @@ func (host *vmContext) SetSystemSCContainer(scContainer vm.SystemSCContainer) er
...
 		return nil
 	}
 
 	return data
 }
 
-// SetStorage saves the key value storage under the address
-func (host *vmContext) SetStorage(key []byte, value []byte) {
-	strAdr := string(host.scAddress)
+// GetStorage get the values saved for a certain key
```
Comment (SebastianMarian) [inline vm/systemSmartContracts/eei.go:86]: gets

File: vm/systemSmartContracts/governance.go [hunk 7]
```diff
@@ -782,30 +781,37 @@ func (g *governanceContract) revokeVotePower(_ *vmcommon.ContractCallInput) vmco
...
+		if err != nil {
+			return 0, err
+		}
+
+		if nodeData.Staked {
+			numStakedNodes++
+		}
+	}
+
+	return auctionData.NumRegistered, nil
```
Comment (SebastianMarian) [inline vm/systemSmartContracts/governance.go:814]: You forgot to set the numStakedNodes to the actionData structure? It is not used anywhere.

File: vm/systemSmartContracts/governance.go (hunk 1 above)
Comment (sasurobert) [reply vm/systemSmartContracts/governance.go:785]: done

File: vm/systemSmartContracts/governance.go (hunk 2 above)
Comment (sasurobert) [reply vm/systemSmartContracts/governance.go:796]: done. thx

File: vm/systemSmartContracts/governance_test.go (hunk 4 above)
Comment (sasurobert) [reply vm/systemSmartContracts/governance_test.go:495]: done

File: vm/systemSmartContracts/governance_test.go (hunk 5 above)
Comment (sasurobert) [reply vm/systemSmartContracts/governance_test.go:498]: done. deleted as no need

File: vm/systemSmartContracts/eei.go (hunk 6 above)
Comment (sasurobert) [reply vm/systemSmartContracts/eei.go:86]: done

File: vm/systemSmartContracts/governance.go (hunk 7 above)
Comment (sasurobert) [reply vm/systemSmartContracts/governance.go:814]: done

**Pull request merged**